src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/transformers/ThreadTransformer.kt

```diff
@@ -57,6 +66,40 @@ internal class ThreadTransformer(
         visitInsn(opcode)
     }
 
+    override fun visitMaxs(maxStack: Int, maxLocals: Int) = adapter.run {
+        if (isThreadRunMethod(methodName, desc)) {
```

Comment:
You can avoid the additional padding by changing the condition to `if (!isThreadRunMethod(..)) { ... }` and calling `return` inside it.